From 7b3b55321db00ec142be02451cbfcf932befcb74 Mon Sep 17 00:00:00 2001
From: chrisber <christian@chrisber.net>
Date: Tue, 9 May 2017 10:44:44 +0200
Subject: [PATCH 1/5] [ADD] pages purge always false

---
 deps/jemalloc/src/pages.c | 23 +----------------------
 1 file changed, 1 insertion(+), 22 deletions(-)

diff --git a/deps/jemalloc/src/pages.c b/deps/jemalloc/src/pages.c
index 83a167f..d4d82ce 100644
--- a/deps/jemalloc/src/pages.c
+++ b/deps/jemalloc/src/pages.c
@@ -146,28 +146,7 @@ bool
 pages_purge(void *addr, size_t size)
 {
 	bool unzeroed;
-
-#ifdef _WIN32
-	VirtualAlloc(addr, size, MEM_RESET, PAGE_READWRITE);
-	unzeroed = true;
-#elif defined(JEMALLOC_HAVE_MADVISE)
-#  ifdef JEMALLOC_PURGE_MADVISE_DONTNEED
-#    define JEMALLOC_MADV_PURGE MADV_DONTNEED
-#    define JEMALLOC_MADV_ZEROS true
-#  elif defined(JEMALLOC_PURGE_MADVISE_FREE)
-#    define JEMALLOC_MADV_PURGE MADV_FREE
-#    define JEMALLOC_MADV_ZEROS false
-#  else
-#    error "No madvise(2) flag defined for purging unused dirty pages."
-#  endif
-	int err = madvise(addr, size, JEMALLOC_MADV_PURGE);
-	unzeroed = (!JEMALLOC_MADV_ZEROS || err != 0);
-#  undef JEMALLOC_MADV_PURGE
-#  undef JEMALLOC_MADV_ZEROS
-#else
-	/* Last resort no-op. */
-	unzeroed = true;
-#endif
+	unzeroed = false;
 	return (unzeroed);
 }
 
-- 
2.7.4


From 2c4046a42c412852921c1aaa81d1b04a3b8755f0 Mon Sep 17 00:00:00 2001
From: chrisber <christian@chrisber.net>
Date: Tue, 9 May 2017 11:36:51 +0200
Subject: [PATCH 2/5] [ADD] prevent rebuild of dependecies

---
 src/Makefile | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index fdbe36a..5752fab 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -163,7 +163,7 @@ persist-settings: distclean
 	echo REDIS_LDFLAGS=$(REDIS_LDFLAGS) >> .make-settings
 	echo PREV_FINAL_CFLAGS=$(FINAL_CFLAGS) >> .make-settings
 	echo PREV_FINAL_LDFLAGS=$(FINAL_LDFLAGS) >> .make-settings
-	-(cd ../deps && $(MAKE) $(DEPENDENCY_TARGETS))
+	#-(cd ../deps && $(MAKE) $(DEPENDENCY_TARGETS))
 
 .PHONY: persist-settings
 
@@ -211,12 +211,13 @@ $(REDIS_CHECK_AOF_NAME): $(REDIS_CHECK_AOF_OBJ)
 	$(REDIS_CC) -c $<
 
 clean:
-	rm -rf $(REDIS_SERVER_NAME) $(REDIS_SENTINEL_NAME) $(REDIS_CLI_NAME) $(REDIS_BENCHMARK_NAME) $(REDIS_CHECK_RDB_NAME) $(REDIS_CHECK_AOF_NAME) *.o *.gcda *.gcno *.gcov redis.info lcov-html
+	$(info "prevent dist  clean")
+	# rm -rf $(REDIS_SERVER_NAME) $(REDIS_SENTINEL_NAME) $(REDIS_CLI_NAME) $(REDIS_BENCHMARK_NAME) $(REDIS_CHECK_RDB_NAME) $(REDIS_CHECK_AOF_NAME) *.o *.gcda *.gcno *.gcov redis.info lcov-html
 
 .PHONY: clean
 
 distclean: clean
-	-(cd ../deps && $(MAKE) distclean)
+	# -(cd ../deps && $(MAKE) distclean)
 	-(rm -f .make-*)
 
 .PHONY: distclean
-- 
2.7.4


From ac3fea6f0e20f667dc14a829a9ba5b844ba0cddd Mon Sep 17 00:00:00 2001
From: chrisber <christian@chrisber.net>
Date: Tue, 9 May 2017 13:49:09 +0200
Subject: [PATCH 3/5] [ADD] without backtrace

---
 src/config.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/config.h b/src/config.h
index 9fd5362..146e36e 100644
--- a/src/config.h
+++ b/src/config.h
@@ -63,7 +63,7 @@
 
 /* Test for backtrace() */
 #if defined(__APPLE__) || (defined(__linux__) && defined(__GLIBC__))
-#define HAVE_BACKTRACE 1
+/*#define HAVE_BACKTRACE 0*/
 #endif
 
 /* MSG_NOSIGNAL. */
-- 
2.7.4


From eda57f01aaa9eff9cb6c96d13308de19f04e0114 Mon Sep 17 00:00:00 2001
From: chrisber <christian@chrisber.net>
Date: Tue, 9 May 2017 14:11:40 +0200
Subject: [PATCH 4/5] [ADD] restrict defines to  uclib

---
 src/config.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/config.h b/src/config.h
index 146e36e..f39edeb 100644
--- a/src/config.h
+++ b/src/config.h
@@ -62,8 +62,8 @@
 #endif
 
 /* Test for backtrace() */
-#if defined(__APPLE__) || (defined(__linux__) && defined(__GLIBC__))
-/*#define HAVE_BACKTRACE 0*/
+#if (defined(__APPLE__) || defined(__linux__) && defined(__UCLIBC__))
+#define HAVE_BACKTRACE 1
 #endif
 
 /* MSG_NOSIGNAL. */
-- 
2.7.4


From be4bea57d97455e7ac6552b832831116c0ca0011 Mon Sep 17 00:00:00 2001
From: chrisber <christian@chrisber.net>
Date: Thu, 11 May 2017 12:10:31 +0200
Subject: [PATCH 5/5] [ADD] prevent default install

---
 src/Makefile | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index 5752fab..9aa5636 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -264,10 +264,11 @@ src/help.h:
 	@../utils/generate-command-help.rb > help.h
 
 install: all
-	@mkdir -p $(INSTALL_BIN)
-	$(REDIS_INSTALL) $(REDIS_SERVER_NAME) $(INSTALL_BIN)
-	$(REDIS_INSTALL) $(REDIS_BENCHMARK_NAME) $(INSTALL_BIN)
-	$(REDIS_INSTALL) $(REDIS_CLI_NAME) $(INSTALL_BIN)
-	$(REDIS_INSTALL) $(REDIS_CHECK_RDB_NAME) $(INSTALL_BIN)
-	$(REDIS_INSTALL) $(REDIS_CHECK_AOF_NAME) $(INSTALL_BIN)
-	@ln -sf $(REDIS_SERVER_NAME) $(INSTALL_BIN)/$(REDIS_SENTINEL_NAME)
+	$(info "prevent default install")
+	# @mkdir -p $(INSTALL_BIN)
+	# $(REDIS_INSTALL) $(REDIS_SERVER_NAME) $(INSTALL_BIN)
+	# $(REDIS_INSTALL) $(REDIS_BENCHMARK_NAME) $(INSTALL_BIN)
+	# $(REDIS_INSTALL) $(REDIS_CLI_NAME) $(INSTALL_BIN)
+	# $(REDIS_INSTALL) $(REDIS_CHECK_RDB_NAME) $(INSTALL_BIN)
+	# $(REDIS_INSTALL) $(REDIS_CHECK_AOF_NAME) $(INSTALL_BIN)
+	# @ln -sf $(REDIS_SERVER_NAME) $(INSTALL_BIN)/$(REDIS_SENTINEL_NAME)
-- 
2.7.4

